readme.md

```diff
@@ -53,48 +50,85 @@ Need to install an ImageData extractor like `get-pixels`
 npm install --save extract-colors get-pixels
 ```
 
-
 ## Usage
 
 ### Browser example
 
 ```js
-import { extractColors } from 'extract-colors'
+import { extractColors } from 'extract-colors';
 
-const src = 'my-image.jpg'
+const src = 'my-image.jpg';
 
-extractColors(src)
-  .then(console.log)
-  .catch(console.error)
+extractColors(src).then(console.log).catch(console.error);
 ```
 
 > You can use different types for `src` param (`String` for a path of image, `HTMLImageElement` or `ImageData`).
 
-
-### Node.js example
+### Node.js example using `get-pixels`
 
 ```js
-const path = require('path')
-const getPixels = require("get-pixels")
-const { extractColors } = require('extract-colors')
+const path = require('path');
+const getPixels = require('get-pixels');
+const { extractColors } = require('extract-colors');
 
-const src = path.join(__dirname, './my-image.jpg')
+const src = path.join(__dirname, './my-image.jpg');
 
 getPixels(src, (err, pixels) => {
-  if(!err) {
-    const data = [...pixels.data]
-    const [width, height] = pixels.shape
+  if (!err) {
+    const data = [...pixels.data];
+    const [width, height] = pixels.shape;
 
-    extractColors({ data, width, height })
-      .then(console.log)
-      .catch(console.log)
+    extractColors({ data, width, height }).then(console.log).catch(console.log);
   }
-})
+});
 ```
 
 > This example use `get-pixels` but you can change the lib.
-> Just send and ImageData object to `extractColors(imageData)`.
+> Just send the ImageData object to `extractColors(imageData)`.
 
+### Node.js example using Typescript and `get-pixels`
+
+```bash
+npm i -D @types/get-pixels
+```
+
+```ts
+import { extractColors } from 'extract-colors';
+import getPixels from 'get-pixels';
+
+type Pixels = {
+  data: Uint8Array;
+  shape: [number, number];
+};
+// ...
+
+async function getPixelsAsync(url: string) {
+  return new Promise(function (resolve, reject) {
+    getPixels(url, function (err, data) {
+      if (err !== null) reject(err);
+      else resolve(data as Pixels);
+    });
+  });
+}
+
+async function getImageColors(imgUrl: string) {
+  const options = {
+    /* ... */
+  };
+  try {
+    const pixels = await getPixelsAsync(imgUrl);
+    if (!pixels) return [];
+
+    const data = [...pixels.data];
+    const [width, height] = pixels.shape;
+    const colors = await extractColors({ data, width, height }, options);
+    return colors;
+  } catch (e) {
+    console.error('Error extracting colors from image', e);
+    return [];
+  }
+}
```

Comment:
In my opinion this part of code is a bit lengthy for the readme, especially since most of the code is used for `getPixels`.

Do you think that a Typescript example is important for users?

If so, perhaps it would be better to add the Typescript examples directly to the doc: https://extract-colors.namide.com/guide/

What do you think about it?

Reply:
apologies for the delay on getting back you you @Namide 

> What do you think about it?

I think that is a good idea! And perhaps a simple addition to the readme like `For Typescript users refer to the docs for detailed example`

> Do you think that a Typescript example is important for users?

Yes, I believe so. The example seems simple but I was only able to piece that together after a good deal of time. I even switched to a few other color libs before coming back. :) Thats why I opted to share, because this lib is very fast and ez to use once the speed bumps are flattened

Reply:
Let me know if you would like me to open a PR to the docs site (i dont see it in this repo)

Reply:
No worries, thanks for your help @dougiefresh49 🙏

> the docs site (i dont see it in this repo)

You're right, it was private. Today I changed the code to be more maintainable.
It's public now: [github.com/Namide/extract-colors-website](https://github.com/Namide/extract-colors-website)

> ...simple addition to the readme like `For Typescript users refer to the docs for detailed example`

Good idea 👍️
In the meantime, this link with Typescript examples can be used: [github.com/Namide/extract-colors-website/tree/main/examples](https://github.com/Namide/extract-colors-website/tree/main/examples)

> Let me know if you would like me to open a PR to the docs site

I you have the time, otherwise I will do it later.
I've don't install linter, please follow the standardjs rules if you want to help.